R/format_output.R

```diff
@@ -0,0 +1,85 @@
+
+
+#' Collapse age groups in output
+#'
+#' Sums over age groups for each compartment
+#'
+#' @param d output data.frame
+#'
+#' @return Output data.frame
+collapse_age <- function(d){
+  d %>%
+    dplyr::group_by(.data$compartment, .data$t, .data$replicate) %>%
+    dplyr::summarise(y =  sum(.data$y)) %>%
+    dplyr::ungroup()
+}
+
+#' Collapse compartments in output
+#'
+#' Sums over simplified groups across compartments
+#'
+#' @param d output data.frame
+#'
+#' @return Output data.frame
+collapse_compartment <- function(d){
+  d %>%
+    dplyr::mutate(group = dplyr::case_when(
+      grepl("IMVGet", .data$compartment) ~ "ICU",
+      grepl("IOxGet", .data$compartment) ~ "hospital",
+      .data$compartment == "n_E2_I" ~ "infections",
+      .data$compartment == "delta_D" ~ "deaths",
+      TRUE ~ .data$compartment)) %>%
+    dplyr::group_by(.data$group, .data$t, .data$replicate) %>%
+    dplyr::summarise(y = sum(.data$y)) %>%
+    dplyr::ungroup() %>%
+    dplyr::rename(compartment = .data$group)
+}
+
+#' Format model output as data.frame
+#'
+#' @param x squire_simulation object
+#' @param var_select Vector of variable names
```

Comment:
Change to "... var_select Vector of compartment names, e.g. \code{c("S", "R")} just so people know that the `var` here means compartments.